flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkSqlParserImplTest.java

```diff
@@ -1336,6 +1380,18 @@ public void testExplainUpsert() {
         this.sql(sql).ok(expected);
     }
 
+    @Test
+    public void testExplainPlanForWithExplainDetails() {
+        String sql = "explain plan for ^json_execution_plan^ upsert into emps1 values (1, 2)";
+        this.sql(sql).fails("(?s).*Encountered \"json_execution_plan\" at line 1, column 18.\n.*");
+    }
```

Comment:
Why add `^` in the sql?

Please use `this.sql(sql).node(new ValidationMatcher(...));` to verify the errors.

Could you also add negative cases where explain with unknown keyword or ddl statement?